metis-enrichment/metis-enrichment-service/src/main/java/eu/europeana/enrichment/service/dao/WikidataAccessDao.java

```diff
@@ -35,19 +37,30 @@
  */
 public class WikidataAccessDao {
 
+  public static final String WIKIDATA_ORGANIZATION_XSL_FILE = "/wkd2org.xsl";
   private static final String SPARQL = "https://query.wikidata.org/sparql";
   private static final int SIZE = 1024 * 1024;
 
   private static final Logger LOGGER = LoggerFactory.getLogger(WikidataAccessDao.class);
 
   private Transformer transformer;
 
-  private File templateFile;
+  private InputStream xslTemplateImputStream;
```

Comment:
I think it is a bad idea to have an input stream as a class variable of a class that is supposed to be a long living object. Also the input stream is never closed and removed. Why not perform the init method immediately upon construction so that you don't need to keep the input stream? Or read the input stream upon construction into a string or byte array?

Reply:
Actually the init() method is called by the constructors. And the stream is closed automatically by the transformer factory class:  transformerFactory.newTransformer(xslt) within the scope of init method. This happens somewhere in the low level implementation of XML (I already verified this in debugger).

Reply:
Firstly, I couldn't find anywhere in the specifications of transformerFactory.newTransformer() that it will close the input stream (or source). So maybe they do, maybe they only do sometimes or under certain conditions, and maybe they will stop doing it in a new version. Please let's just be safe and make sure the input stream is closed.

Secondly: I don't know what is inside the input stream that is passed. There may be a byte array that is no longer needed and that will only be freed once the stream is garbage collected. 

I still don't see why the input stream cannot just be passed directly to the init() method. But if that really cannot be done, may I suggest the following alternative? 

Instead of passing an input stream during construction, we pass an input stream supplier: java.util.function.Supplier<InputStream>. Then in the init() method we create the input stream (using the supplier), read it and close it, and nothing is kept in memory. The callers are then completely responsible for using it properly (i.e. they should not simply pass a wrapper around an input stream but actually specify how the stream is to be created).

Any thoughts?

Reply:
Well I don't expect the JDK rt.jar will be changed so that they don't close streams anymore in the future versions. 
Now the default constructur is used by the code, 
this.xslTemplateImputStream = getClass().getResourceAsStream(WIKIDATA_ORGANIZATION_XSL_FILE);
So .. the input stream is the connected to the /wkd2org.xsl which is available in resources.

I can pass the inputStream to the init() method if you consider this to be better.